test/Microsoft.AspNetCore.SignalR.Tests/EndToEndTests.cs

```diff
@@ -45,5 +49,42 @@ public EndToEndTests(ServerFixture serverFixture)
                 await ws.CloseAsync(WebSocketCloseStatus.Empty, "", CancellationToken.None);
             }
         }
+
+        [Fact]
+        public async Task CheckConnection()
+        {
+            const string message = "Major Key";
+            var baseUrl = _serverFixture.BaseUrl;
+            var loggerFactory = new LoggerFactory();
+
+            using (var httpClient = new HttpClient())
+            {
+                var transport = new LongPollingTransport(httpClient, loggerFactory);
+                using (var connection = await Connection.ConnectAsync(new Uri(baseUrl + "/echo"), transport, httpClient, loggerFactory))
+                {
+                    await connection.Output.WriteAsync(new Message(
+                        ReadableBuffer.Create(Encoding.UTF8.GetBytes(message)).Preserve(),
+                        Format.Text));
+
+                    var received = await ReceiveMessage(connection).OrTimeout();
+                    Assert.Equal(message, received);
+                }
+            }
+        }
+
+
+        private static async Task<string> ReceiveMessage(Connection connection)
+        {
+            Message message;
+            while (await connection.Input.WaitToReadAsync())
+            {
+                if (connection.Input.TryRead(out message))
+                {
```

Comment:
Indeed it should, you'll need a `using` block inside the `if`. Technically it doesn't need to be disposed in this scenario, but we want to ensure we're using the right pattern for when we introduce pooling (disposing would return message buffers to the pool)